testing/deprecated_test.py

```diff
@@ -43,15 +45,19 @@ def test_terminal_reporter_writer_attr(pytestconfig):
     except ImportError:
         pass
     terminal_reporter = pytestconfig.pluginmanager.get_plugin("terminalreporter")
-    expected_tw = terminal_reporter._tw
-
-    with pytest.warns(pytest.PytestDeprecationWarning):
-        assert terminal_reporter.writer is expected_tw
-
-    with pytest.warns(pytest.PytestDeprecationWarning):
-        terminal_reporter.writer = expected_tw
-
-    assert terminal_reporter._tw is expected_tw
+    original_tw = terminal_reporter._tw
+
+    with pytest.warns(pytest.PytestDeprecationWarning) as cw:
+        assert terminal_reporter.writer is original_tw
+    assert len(cw) == 1
+    assert cw[0].filename == __file__
+
+    new_tw = copy.copy(original_tw)
+    with pytest.warns(pytest.PytestDeprecationWarning) as cw:
+        terminal_reporter.writer = new_tw
```

Comment:
the copy has to be taken out of the global plugin again
key reason why i didn't brother with a different object there

after all this is code that's to stay there just until its removed again